src/internal/obj/amazon_client.go

```diff
@@ -301,10 +301,14 @@ func (c *amazonClient) transformError(err error, objectPath string) error {
 	if strings.Contains(err.Error(), "Not Found") {
 		return pacherr.NewNotExist(c.bucket, objectPath)
 	}
-	var awsErr awserr.Error
-	if !errors.As(err, &awsErr) {
+	awsErr, ok := err.(awserr.Error)
```

Comment:
pretty sure we should be using `errors.As` instead of casting as a general rule since we have wrapped errors around

Reply:
I think errors.Is and errors.As are for concrete types.  For interfaces (which `awserr.Error` is), you are supposed to use a normal interface cast.

Reply:
The request canceled case should break if they start returning awserr wrapped in anything.